frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/virtual-hardware-tab/virtual-hardware-tab.tsx

```diff
@@ -0,0 +1,214 @@
+  pvcs: FirehoseResult<K8sResourceKind[]>,
+): VMWizardStorageBundle[] => {
+  const pvcLookup = createLookup(pvcs, getName);
+
+  return storages
+    .filter((storage) => VHW_TYPES.has(storage.diskWrapper.getType()))
+    .map((wizardStorageData) => {
+      const {
+        diskWrapper,
+        volumeWrapper,
+        dataVolumeWrapper,
+        persistentVolumeClaimWrapper,
+      } = wizardStorageData;
+
+      const pvc = pvcLookup[volumeWrapper.getPersistentVolumeClaimName()];
+
+      const combinedDisk = new CombinedDisk({
+        diskWrapper,
+        volumeWrapper,
+        dataVolumeWrapper,
+        persistentVolumeClaimWrapper:
+          persistentVolumeClaimWrapper || (pvc && PersistentVolumeClaimWrapper.initialize(pvc)),
+        isNewPVC: !!persistentVolumeClaimWrapper,
+        pvcsLoading: !isLoaded(pvcs),
+      });
+
+      return {
+        wizardStorageData,
+        source: combinedDisk.getCDROMSourceValue(),
+        content: combinedDisk.getContent(),
+        storageClass: combinedDisk.getStorageClassName(),
+      };
+    });
+};
+
+const VirtualHardwareTabFirehose: React.FC<VirtualHardwareTabFirehoseProps> = ({
+  wizardReduxID,
+  isLocked,
+  setTabLocked,
+  removeStorage,
+  storages,
+  persistentVolumeClaims,
+}) => {
+  const virtualStorages = getVirtualStoragesData(storages, persistentVolumeClaims);
+  const showStorages = virtualStorages.length > 0;
+  const disableAddCD = isLocked || virtualStorages.length > 1;
+  const availableCDName = getAvailableCDName(storages.map((storage) => storage.disk));
+  const withProgress = wrapWithProgress(setTabLocked);
+  const diskWrapper = DiskWrapper.initializeFromSimpleData({
+    name: availableCDName,
+    type: DiskType.CDROM,
+    bus: DiskBus.VIRTIO,
+  });
+
+  const addButton = (
+    <Button
+      variant={ButtonVariant.link}
+      isDisabled={disableAddCD}
+      icon={<PlusCircleIcon />}
+      id="attach-cdrom"
+      className="pf-m-link--align-left virtual-hardware-tab-add-btn"
+      onClick={() =>
+        withProgress(
+          vmWizardStorageModalEnhanced({
+            storage: {
+              diskWrapper,
+              type: VMWizardStorageType.UI_INPUT,
+            },
+            blocking: true,
+            wizardReduxID,
+          }).result,
+        )
+      }
+    >
+      {ATTACH_CD}
+    </Button>
+  );
+
+  return (
+    <div className="kubevirt-create-vm-modal__virtual-hardware-tab-container">
+      <Title headingLevel="h5" size="lg">
+        Virtual Hardware
+      </Title>
+      <Title className="virtual-hardware-tab-cd-title" headingLevel="h4" size="md">
+        CD-ROMs
+      </Title>
+      {showStorages && (
+        <>
+          <div className="virtual-hardware-tab-main">
+            <VMCDsTable
+              columnClasses={cdTableColumnClasses}
+              data={virtualStorages}
+              customData={{ isDisabled: disableAddCD, withProgress, removeStorage, wizardReduxID }}
```

Comment:
why is the editing and deleting disabled once we reach 2 cds?

Reply:
@suomiy thats a design decision. A bit controversial I agree, but a reasoning is that it is not needed to have more than 2 - you can have 1 for OS and 1 for drivers but there is no need to have a 3th. ...aaaand we should expose only things which are actually needed...

Reply:
Ah, sorry, I've misunderstood your comment. Also the delete and edit is disabled here, so it is incorrect I agree. Only the add has to be disabled.

Reply:
fixed

Reply:
it should still reference `isLocked`